### PR TITLE
Refactor Cypress Tests, Resolve Mochawesome and Slack Reporter Issues, updated README.ME and Add .gitignore

### PR DESCRIPTION
**Pull request by FahadDarw**
**Pull Request Description:**

### Changes Made:
- Refactored Cypress tests for improved readability and maintainability.
- Addressed issues with Mochawesome and Slack reporter integration.
- Added a `.gitignore` file to exclude unnecessary files from version control.

### Detailed Changes:

**1. Refactored Cypress Tests:**
   - [x] Restructured Cypress tests to enhance code organization and readability.
   - [x] Improved modularity of test scenarios for easier maintenance.

**2. Mochawesome and Slack Reporter Fixes:**
   - [x] Resolved issues affecting Mochawesome reporting, ensuring accurate and reliable test result output.
   - [x] Fixed integration with the Slack reporter to provide comprehensive notifications and insights.

**3. Added .gitignore:**
   - [x] Introduced a `.gitignore` file to exclude specific files and directories from version control.
   - [x] Ignored files:
      - `Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/cypress.env.json`
      - `Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/cypress.env`
      - `Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/cypress.json`
      - `Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/cypress/reports/mocha/`

### Additional Information:

**1. Testing:**
   - [x] Conducted thorough testing to ensure the reliability and effectiveness of the changes.

**3. Notes:**
   - [x] Any additional notes to README.MD file regarding the way cypress now run by using npm run cy:run for headless and npm run cy:open for headed.

### Reviewer Guidance:

- **Testing Focus:**
   - [x] Pay attention to the changes made in Cypress tests and verify their effectiveness.

- **Configuration Updates:**
   - [x] Review the changes in Mochawesome and Slack reporter configurations.

- **.gitignore Addition:**
   - [x] Confirm that the new `.gitignore` file appropriately excludes unnecessary files.

### CODE REVIEW

File: .github/workflows/build-and-push-image.yml [hunk 1]
```diff
@@ -173,9 +173,25 @@ jobs:
...
           path: Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/cypress/screenshots
-          
+
+      - name: Generate report
+        if: always()
+        run: |
+            mkdir mochareports
+            npm run generate:html:report    
+
+            - name: Upload report
```
Comment (cshnimble) [inline .github/workflows/build-and-push-image.yml:183]: Check the formatting of this, I suspect it'll cause the pipeline to fail due to incorrect indentation

File: Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json [hunk 2]
```diff
@@ -4,10 +4,15 @@
...
-    "cy:run": "cypress run",
-    "lint": "eslint . --ext .cy.js,.js,.jsx,.ts,.tsx --ignore-path .gitignore",
-    "cy:notify": "cypress-slack-reporter"
+    "cy:open": "cypress open --browser edge",
+    "cy:run": "npm run clean:reports && cypress run --browser edge",
+    "cy:zap": "npm run clean:reports && cypress run --browser electron",
+    "cy:notify": "cypress-slack-reporter",
+    "clean:reports": "rm -rf cypress/reports/mocha/*.json",
+    "combine:reports": "mochawesome-merge  cypress/reports/mocha/*.json> mochareports/report.json",
+    "create:html:report": "marge  mochareports/report.json -f report -o mochareports",
```
Comment (cshnimble) [inline Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json:13]: Should this be merge?

**Review by cshnimble — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-push-image.yml (hunk 1 above)
Comment (FahadDarw) [reply .github/workflows/build-and-push-image.yml:183]: I checked it, it seems to be okay. but I can reformat on the next push if there is any problems.

File: Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json (hunk 2 above)
Comment (mikestock-nimble) [reply Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json:13]: Hi Chris,

It is actually marge, which is hilarious :D

Comment (FahadDarw) [reply Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json:13]: its actually marge-json :P but it is what it is :P

**Review by mikestock-nimble — APPROVED** [review]

File: Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json (hunk 2 above)
Comment (cshnimble) [reply Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json:13]: You learn something new every day!

**Review by cshnimble — APPROVED** [review]

**Review by dangood84 — APPROVED** [review]

**Pull request merged**